client/components/domains/register-domain-step/index.jsx

```diff
@@ -365,11 +374,19 @@ class RegisterDomainStep extends React.Component {
 
 	renderPaginationControls() {
 		const isKrackenUi = config.isEnabled( 'domains/kracken-ui/pagination' );
-		if ( ! isKrackenUi || this.state.searchResults === null ) {
+		if ( ! isKrackenUi ) {
 			return null;
 		}
 
-		const isLoading = this.state.loadingResults || this.state.loadingNextPageResults;
+		if ( this.state.searchResults === null ) {
+			return null;
+		}
+
+		if ( this.state.pageNumber >= MAX_PAGES ) {
+			return null;
+		}
+
+		const isLoading = this.state.loadingResults;
```

Comment:
OK, sorry if it's a stupid question but why would we go for deconstructing vs assignment in case it's a single value we want to get from the state? What are the benefits?

Reply:
Ah... ignore that :D We're using several state variables in this code. Will fix it.